h5netcdf/core.py

```diff
@@ -234,8 +245,52 @@ def __init__(self, parent, name):
                     if k.startswith('_nc4_non_coord_'):
                         var_name = k[len('_nc4_non_coord_'):]
                     self._variables.add(var_name)
+
+        if has_existing_unlimited_dims is True:
+            self._determine_current_dimension_sizes()
+
         self._initialized = True
 
+    def _determine_current_dimension_sizes(self):
+        """
+        Helper method to determine the current size of all unlimited
+        dimensions.
+        """
+        def _find_dim(h5group, dim):
+            if dim not in h5group:
+                return _find_dim(h5group.parent, dim)
+            return h5group[dim]
+
+        for d in self.dimensions:
```

Comment:
Would it make sense to only iterate over `self._dim_sizes.maps[-1]` here instead of `self.dimensions`?

This would only pick up dimensions newly newly defined in this group (in the file), not a parent group, so it would avoid some redundant work.

And if the file has new dimensions and is not yet flushed to disk, those new dimensions wouldn't be iterated over, either. So we wouldn't need the boolean `has_existing_unlimited_dims`.

Reply:
That is not possible as more then one dimension might be set during the init. I found a different way though: The current size is now determined per dimension and only when it is necessary at the time the dimension is set.

Reply:
(Correction, I was actually thinking of `self._dim_sizes.maps[0]`.)

That should pull out the mapping with everything set in the most recent "layer" of the ChainMap, which means everything set in this `Group`: https://docs.python.org/3/library/collections.html#collections.ChainMap

I'll take a look at your solution.